server/lib/two-factor-authentication/fido-metadata.ts

```diff
@@ -0,0 +1,99 @@
+import crypto from 'crypto';
+
+import { Mutex } from 'async-mutex';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import moment from 'moment';
+
+import { reportErrorToSentry } from '../sentry';
+
+import sourceCachedMetadata from './cached-metadata.json';
+
+export type Metadata = {
+  nextUpdate: string;
+  entries: MetadataEntry[];
+};
+
+export type MetadataEntry = {
+  aaguid?: string;
+  metadataStatement?: MetadataStatement;
+  statusReports: {
+    status: string;
+  }[];
+};
+
+type MetadataStatement = {
+  description?: string;
+  icon?: string;
+};
+
+const mutex = new Mutex();
+let cachedMetadata: Metadata = sourceCachedMetadata;
+let nextUpdate = cachedMetadata.nextUpdate;
+let cachedEntriesByAaguid: Record<string, MetadataEntry> = cachedMetadata.entries.reduce(
+  (acc, current) => {
+    if (!current.aaguid) {
+      return acc;
+    }
+
+    return {
+      ...acc,
+      [current.aaguid]: {
+        ...current,
+      },
+    };
+  },
+  {} as Record<string, MetadataEntry>,
+);
+
+export async function downloadFidoMetadata(): Promise<Metadata> {
+  const fidoAlianceMetadataUrl = 'https://mds3.fidoalliance.org';
+
+  const response = await axios.get<string>(fidoAlianceMetadataUrl);
```

Comment:
Could we use node-fetch? while it's not formalised in an RFC, we're trying to not depend on axios for simple things.